Update Kubernetes Access docs (#5865)
gravitational/teleport: docs/pages/kubernetes-access/guides/cicd.mdx

---
title: Teleport Kubernetes Access for CI/CD
description: Short Lived Certs CI/CD systems to Kubernetes RBAC with Teleport
---

## Short Lived Certs for Kubernetes CI/CD

CI/CD tools like Jenkins can use short-lived certificates to talk to Kubernetes
API.

> Non interactive local Teleport user, then exporting

Reviewer: Reads weird, esp. "then exporting" piece seems out of place.

---
title: Teleport Kubernetes Access for CI/CD
description: Short Lived Certs CI/CD systems to Kubernetes RBAC with Teleport
---

## Short Lived Certs for Kubernetes CI/CD

> CI/CD tools like Jenkins can use short-lived certificates to talk to Kubernetes

Reviewer 2: ```suggestion
CI/CD tools like Jenkins can use short-lived certificates to talk to the Kubernetes
```

API.

Create a non-interactive local Teleport user. Non-interactive users
don't have username or password defined and can't login, but can be used
to create credentials.

```yaml
kind: role
version: v3
metadata:
  name: robot
spec:
  # allow section declares a list of resource/verb combinations that are
  # allowed for the users of this role. by default nothing is allowed.

>   allow:

Reviewer 2: ```suggestion
  # This field is used for SSH logins. You have to keep 'logins' as a non-empty random value
  # for Kubernetes to work until we fix it.
  allow:
```

    logins: ['keep any value here']
    # a list of kubernetes groups to assign
    kubernetes_groups: ['system:masters']
---
kind: user
version: v3
metadata:
  name: jenkins
spec:
  roles:
  - robot
```

> Generate a kubeconfig using `jenkins` user and it's roles using [`tctl auth sign`](cli-docs.md#tctl-auth-sign):

Reviewer 2: ```suggestion
Generate a kubeconfig using the `jenkins` user and its roles using [`tctl auth sign`](cli-docs.md#tctl-auth-sign):
```

```bash
# Create a new local user for Jenkins

> $ tctl users add jenkins

Reviewer 2: ```suggestion
$ tctl users add jenkins --roles=robot
```

# Creates a token for 25hrs
$ tctl auth sign --user=jenkins --format=kubernetes --out=kubeconfig --ttl=25h

  The credentials have been written to kubeconfig

$ cat kubeconfig
  apiVersion: v1
  clusters:
  - cluster:
      certificate-authority-data: LS0tLS1CRUdJTiBDRVJUSUZ....
# This kubeconfig can now be exported and will provide access to the automation tooling.

# Uses kubectl to get pods, using the provided kubeconfig.
$ kubectl --kubeconfig /path/to/kubeconfig get pods
```

<Admonition type="tip" title="Short-Lived Certs">
Short lived certificates expire in hours or minutes. You don't have to revoke
them if the host gets compromised.
Generate a new kubeconfig every hour using `tctl` or [API](../api-reference.md)

> and publish it to secrets storage, like [AWS](https://aws.amazon.com/secrets-manager/) or

Reviewer 2: ```suggestion
and publish it to secret storage, like [AWS](https://aws.amazon.com/secrets-manager/) or
```

> [GCP](https://cloud.google.com/secret-manager) secrets managers.

Reviewer 2: ```suggestion
[GCP](https://cloud.google.com/secret-manager) secret managers.
```
